docs/reference/index-modules/command-line-tools.asciidoc

```diff
@@ -0,0 +1,213 @@
+!   WARNING: Elasticsearch MUST be stopped before running this tool   !
+!                                                                     !
+!   WARNING:    Documents inside of translog files will be lost       !
+!                                                                     !
+!   WARNING:          The following files will be DELETED!            !
+!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+--> data/nodes/0/indices/P45vf_YQRhqjfwLMUvSqDw/0/translog/translog-41.ckp
+--> data/nodes/0/indices/P45vf_YQRhqjfwLMUvSqDw/0/translog/translog-6.ckp
+--> data/nodes/0/indices/P45vf_YQRhqjfwLMUvSqDw/0/translog/translog-37.ckp
+--> data/nodes/0/indices/P45vf_YQRhqjfwLMUvSqDw/0/translog/translog-24.ckp
+--> data/nodes/0/indices/P45vf_YQRhqjfwLMUvSqDw/0/translog/translog-11.ckp
+
+Continue and DELETE files? [y/N] y
+Reading translog UUID information from Lucene commit from shard at [data/nodes/0/indices/P45vf_YQRhqjfwLMUvSqDw/0/index]
+Translog Generation: 3
+Translog UUID      : AxqC4rocTC6e0fwsljAh-Q
+Removing existing translog files
+Creating new empty checkpoint at [data/nodes/0/indices/P45vf_YQRhqjfwLMUvSqDw/0/translog/translog.ckp]
+Creating new empty translog at [data/nodes/0/indices/P45vf_YQRhqjfwLMUvSqDw/0/translog/translog-3.tlog]
+
+Marking index with the new history uuid : TAUddBstTciV9wAiA6sKFA
+Changing allocation id ceU7CskxT4yRw4M-GLO1mg to nFa2DcCsSlady4LlnJeaEQ
+You should run follow command to apply allocation id changes:
+
+$ curl -XPOST 'http://localhost:9200/_cluster/reroute' -d '
+{
+  "commands" : [
+    {
+      "allocate_stale_primary" : {
+        "index" : "twitter",
+        "shard" : 0,
+        "node" : "pAfJBgAAQACIfI2M_____w",
+        "accept_data_loss" : true
+      }
+    }
+  ]
+}'
+Done.
+--------------------------------------------------
+
+=== What to do if the Lucene index becomes corrupted?
+
+In a similar cases the index on a shard copy can become corrupted.
+Like in case with corrupted translog when index corruption is detected by Elasticsearch due
+to mismatching checksums, Elasticsearch will fail that shard copy and refuse to use that copy of the data.
+If there are other copies of the shard available then Elasticsearch will automatically recover from one of
+them using the normal shard allocation and recovery mechanism.
+
+In order to remove corrupted segments use `remove-corrupted-segments` subcommand:
+It writes a new segments file that removes reference to problematic (corrupted) Lucene segments if there is
+no copy of the data from which Elasticsearch can recover successfully.
+
+Highly recommended to make a complete backup of your index before using this to remove corrupted documents
+from your index!
+
+* you should specify index name with `--index` and shard id `--shard-id`
+* or specify the full path to corrupted translog with the `-d` option
+
+You can get an overview of the corruption with `--dry-run` option :
+
```

Comment:
I'd be inclined to break the dry-run option out into it's own section:
[float]
[[list-corrupted-segments]]
=== Listing corrupted index segments

To see what segments are corrupted and will be dropped, use the `--dry-run` option with the `remove-corrupted-segments` subcommand. There are two ways to specify the shard:

* Specify the index name and shard name with the `--index` and `--shard-id` options.
* Use the `-d` option to specify the full path to the corrupted segment.